### PR TITLE
changes needed for inventory refactor

### PR DESCRIPTION
**Pull request by PaulRitter**
requires https://github.com/space-wizards/space-station-14/pull/5807

### CODE REVIEW

File: Robust.Shared/Containers/BaseContainer.cs [hunk 1]
```diff
@@ -101,9 +101,24 @@ public virtual bool CanInsert(EntityUid toinsert, IEntityManager? entMan = null)
...
+            //raise events
+            var insertAttemptEvent = new ContainerIsInsertingAttemptEvent(this, toinsert);
+            entMan.EventBus.RaiseLocalEvent(Owner, insertAttemptEvent);
+            if (insertAttemptEvent.Cancelled)
+                return false;
+
+            var gettingInsertedAttemptEvent = new ContainerGettingInsertedAttemptEvent(this, toinsert);
+            entMan.EventBus.RaiseLocalEvent(toinsert, gettingInsertedAttemptEvent);
+            if (gettingInsertedAttemptEvent.Cancelled)
+                return false;
```
Comment (mirrorcult) [inline Robust.Shared/Containers/BaseContainer.cs:119]: huh?

Comment (mirrorcult) [reply Robust.Shared/Containers/BaseContainer.cs:119]: nvm these are on two separte entities

File: Robust.UnitTesting/Server/GameObjects/Components/Container_Test.cs [hunk 2]
```diff
@@ -326,9 +326,7 @@ public override void Shutdown()
 
             public override bool CanInsert(EntityUid toinsert, IEntityManager? entMan = null)
             {
```
Comment (moonheart08) [inline Robust.UnitTesting/Server/GameObjects/Components/Container_Test.cs:328]: Why does this take an entitymanager ref and then never use it?

**Review by moonheart08 — CHANGES_REQUESTED** [review]

File: Robust.UnitTesting/Server/GameObjects/Components/Container_Test.cs (hunk 2 above)
Comment (PaulRitter) [reply Robust.UnitTesting/Server/GameObjects/Components/Container_Test.cs:328]: good point!

**Review by metalgearsloth — APPROVED** [review]

**Review by ElectroJr — APPROVED** [review]

**Review by metalgearsloth — APPROVED** [review]

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**